iceoryx_hoofs/posix/filesystem/source/file.cpp

```diff
@@ -0,0 +1,438 @@
+// Copyright (c) 2023 by Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include "iox/file.hpp"
+#include "iceoryx_hoofs/posix_wrapper/posix_call.hpp"
+#include "iceoryx_platform/attributes.hpp"
+#include "iceoryx_platform/errno.hpp"
+#include "iceoryx_platform/fcntl.hpp"
+#include "iceoryx_platform/stdio.hpp"
+#include "iox/filesystem.hpp"
+
+namespace iox
+{
+expected<File, FileCreationError> FileBuilder::create(const FilePath& name) noexcept
+{
+    switch (m_open_mode)
+    {
+    case posix::OpenMode::OPEN_EXISTING:
+        return this->open(name);
+    case posix::OpenMode::PURGE_AND_CREATE:
+        if (File::remove(name).has_error())
+        {
+            IOX_LOG(ERROR) << "Unable to purge and open file \"" << name.as_string()
+                           << "\" since the file could not be removed";
+            return iox::error<FileCreationError>(FileCreationError::CannotBePurged);
+        }
+        IOX_FALLTHROUGH;
+    case posix::OpenMode::EXCLUSIVE_CREATE:
+        return this->open_impl(true, name);
```

Comment:
What do you think of using constexpr like `PRINT_ERROR_ON_NON_EXISTING_FILE{true}`/`DO_NOT_PRINT_ERROR_ON_NON_EXISTING_FILE{false}`. It would make the code more readable. An enum would also work but since this is not a public API it is not that important.

On a second thought. It seems `open` just calls `open_impl` with `true`. All `open_impl` with `true` could therefore be replace by `open` and for the false case the constexpr could be used ... and in `open` also a constexpr for `true`